planner/core/logical_plan_builder.go

```diff
@@ -5612,3 +5734,284 @@ func containDifferentJoinTypes(preferJoinType uint) bool {
 	}
 	return cnt > 1
 }
+
+func (b *PlanBuilder) buildCte(ctx context.Context, cte *ast.CommonTableExpression, isRecursive bool) (p LogicalPlan, err error) {
+	if isRecursive {
+		saveCheck := b.buildingRecursivePartForCTE
+		b.buildingRecursivePartForCTE = false
```

Comment:
Why we set this to false when go into `buildRrrecursiveCTE`?

Reply:
// buildingRecursivePartForCTE likes a stack. We save it before building a recursive CTE and restore it after building.
		// We need a stack because we need to handle the nested recursive CTE. And buildingRecursivePartForCTE indicates the innermost CTE.

I add a comment for it.